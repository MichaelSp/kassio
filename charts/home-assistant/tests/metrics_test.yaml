suite: Metrics Suite
templates:
  - servicemonitor.yaml
chart:
  version: 0.1.0
  appVersion: 1.2.3
tests:
  - it: has metrics
    set:
      metrics:
        enabled: true
        prometheusRule:
          enabled: false
        serviceMonitor:
          authorization:
            credentials:
              name: home-assistant-scrape-secrets
              key: bearer-token
    asserts:
      - matchSnapshot: { }