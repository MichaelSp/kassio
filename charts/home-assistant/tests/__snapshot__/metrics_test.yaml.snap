has metrics:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
    spec:
      endpoints:
        - authorization: |2-
            credentials:
              key: bearer-token
              name: home-assistant-scrape-secrets
          interval: 1m
          path: /api/prometheus
          port: http
          scrapeTimeout: 30s
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: home-assistant
